modules/cudf/src/data_frame.ts

```diff
@@ -847,6 +848,33 @@ export class DataFrame<T extends TypeMap = any> {
     }
   }
 
+  /**
+   * Compute the trigonometric cosine inverse for all NumericSeries in the DataFrame
+   *
+   * @returns A DataFrame with the operation performed on all NumericSeries
+   * @example
+   * ```typescript
+   * import {DataFrame, Series}  from '@rapidsai/cudf';
+   *
+   * const df = new DataFrame({
+   *  a: Series.new({type: new Int8, data: [-3, 0, 3]});
+   * });
+   * df.acos();
+   * // return {
+   * //    a: [0, 1, 0],
+   * // }
+   * ```
+   */
+  acos(): DataFrame<T> {
+    return new DataFrame(this.names.reduce((map, name) => ({
+                                             ...map,
+                                             [name]: this.get(name) instanceof NumericSeries
+                                                       ? Series.new(this._accessor.get(name).acos())
+                                                       : Series.new(this._accessor.get(name))
+                                           }),
+                                           {} as SeriesMap<T>));
+  }
```

Comment:
This is lovely. I might personally find it a little more friendly to read more like this:
```typescript
  acos(): DataFrame<T> {
    const mapper = (map: SeriesMap<T>, name: string) => {
      return {
        ...map, [name]: this.get(name) instanceof NumericSeries
                  ? Series.new(this._accessor.get(name).acos())
                  : Series.new(this._accessor.get(name))
      }
    };
    return new DataFrame(this.names.reduce(mapper, {} as SeriesMap<T>));
  }
```

Which also makes me wonder if maybe there could even be a factory function to generate the mapper, so that all the methods could just be something like
```typescript
 return new DataFrame(this.names.reduce(this._make_mapper("acos"), {} as SeriesMap<T>));
```
but @trxcllnt may think that is too clever by half :grinning:

Reply:
Yeah I'm open to whether or not this should be condensed even more. I was considering it as something similar is done in the tests themselves...

https://github.com/rapidsai/node-rapids/blob/781d9f12465ce0d570e14e91cf5f67794b430f3b/modules/cudf/test/series/unaryop/int8-tests.ts#L37

Reply:
>  I'm assuming the only way to implement such a function would be to pass in a string of the method name

AFAIK, which would definitely be the biggest drawback. But 10-100s of lines of boilerplate has its own drawbacks as well, so it's definitely a trade-off. @trxcllnt might have some clever way to make the typing situation better, though.

Reply:
Might have been overthinking this, I pushed some new examples of how `acos` could potentially look. I think the current definition of `acos()` is probably best. (Unless there is something else I can abstract away)